xscen/diagnostics.py

```diff
@@ -97,535 +110,251 @@ def _invert_unphysical_temperatures(
-    )
 
-    ps = sim.plot(
-        ax=axsm,
-        vmin=vmin,
-        vmax=-vmin,
-        cmap=cmbias,
-        add_colorbar=False,
-        transform=crs.PlateCarree(),
-    )
-    scen.plot(
-        ax=axsc,
-        vmin=vmin,
-        vmax=-vmin,
-        cmap=cmbias,
-        add_colorbar=False,
-        transform=crs.PlateCarree(),
-    )
-    if op == "distance":
-        diff = abs(sim - scen)
-    elif op == "improvement":
-        diff = abs(sim) - abs(scen)
-    elif op == "difference":  # op == 'diff':
-        diff = sim - scen
+    if isinstance(properties, (str, Path)):
+        logger.debug("Loading properties module.")
+        module = load_xclim_module(properties)
+        properties = module.iter_indicators()
+    elif hasattr(properties, "iter_indicators"):
+        properties = properties.iter_indicators()
+
+    try:
+        N = len(properties)
+    except TypeError:
+        N = None
     else:
-        raise NotImplementedError(
-            "Operation not implemented. Try 'distance', 'improvement' or 'difference' instead."
+        logger.info(f"Computing {N} properties.")
+
+    # select periods for ref_measure
+    if (
+        dref_for_measure is not None
+        and period is not None
+        and "time" in dref_for_measure
+    ):
+        dref_for_measure = dref_for_measure.sel(
+            {"time": slice(str(period[0]), str(period[1]))}
         )
-    pc = diff.plot(
-        ax=axim,
-        robust=True,
-        cmap=cmimpr,
-        center=0,
-        add_colorbar=False,
-        transform=crs.PlateCarree(),
+
+    if unstack:
+        ds = unstack_fill_nan(ds)
+
+    if change_units_arg:
+        ds = change_units(ds, variables_and_units=change_units_arg)
+
+    prop = xr.Dataset()  # dataset with all properties
+    meas = xr.Dataset()  # dataset with all measures
+    for i, ind in enumerate(properties, 1):
+        if isinstance(ind, tuple):
+            iden, ind = ind
+        # Make the call to xclim
+        logger.info(f"{i} - Computing {iden}.")
+        out = ind(ds=ds)
+        vname = out.name
+        prop[vname] = out
+        if period:
+            prop[vname].attrs["period"] = f"{period[0]}-{period[1]}"
+
+        # calculate the measure if a reference dataset is given for the measure
+        if dref_for_measure and vname in dref_for_measure:
+            meas[vname] = ind.get_measure()(
+                sim=prop[vname], ref=dref_for_measure[vname]
+            )
+            # create a merged long_name
+            prop_ln = prop[vname].attrs.pop("long_name", "").replace(".", "")
+            meas_ln = meas[vname].attrs.pop("long_name", "").lower()
+            meas[vname].attrs["long_name"] = f"{prop_ln} {meas_ln}"
+
+    for ds1 in [prop, meas]:
+        ds1.attrs = ds.attrs
+        ds1.attrs["cat:xrfreq"] = "fx"
+        ds1.attrs.pop("cat:variable", None)
+        ds1.attrs["cat:frequency"] = "fx"
+        ds1.attrs["cat:timedelta"] = "NAN"
```

Comment:
```suggestion
        ds1.attrs["cat:timedelta"] = "NAN"
```
We shouldn't have `cat:timedelta` anymore.

Reply:
True, oops!